rules/rule_manager.go

```diff
@@ -1,33 +1,40 @@
 package rules
 
 import (
+	"sort"
 	"strings"
 )
 
 type ruleManager struct {
 	constraints        map[string]constraint
 	currentIndex       int
 	rulesBySlashCount  map[int]map[DynamicRule]int
-	prefixes           map[string]string
-	rules              []DynamicRule
+	prefixes           map[string]ruleMgrRuleOptions
+	rules              map[DynamicRule]uint
 	enhancedRuleFilter bool
 }
 
+type ruleMgrRuleOptions struct {
+	crawlerOnly bool
+	priority    uint
+}
+
 func newRuleManager(constraints map[string]constraint, enhancedRuleFilter bool) ruleManager {
 	rm := ruleManager{
 		rulesBySlashCount:  map[int]map[DynamicRule]int{},
-		prefixes:           map[string]string{},
+		prefixes:           map[string]ruleMgrRuleOptions{},
 		constraints:        constraints,
 		currentIndex:       0,
-		rules:              []DynamicRule{},
+		rules:              map[DynamicRule]uint{},
 		enhancedRuleFilter: enhancedRuleFilter,
 	}
 	return rm
 }
 
-func (rm *ruleManager) getStaticRules(key string, value *string) map[staticRule]int {
+func (rm *ruleManager) getStaticRules(key string, value *string) (map[staticRule]int, []staticRule) {
 	slashCount := strings.Count(key, "/")
 	out := make(map[staticRule]int)
+	prioritized := make(map[staticRule]uint)
 	rules, ok := rm.rulesBySlashCount[slashCount]
```

Comment:
Wonder if we can change this to be a sorted slice by priority which would be done upfront in AddRule.   Seems like a few parts of this are pointing to trying to find/use a sorted map implementation.